core/pkg/artifacts/saver.go

```diff
@@ -138,6 +138,41 @@ func (as *ArtifactSaver) createManifest(
 	return response.GetCreateArtifactManifest().ArtifactManifest, nil
 }
 
+func (as *ArtifactSaver) updateManifest(
+	artifactManifestId string, manifestDigest string,
+) (attrs gql.UpdateArtifactManifestUpdateArtifactManifestUpdateArtifactManifestPayloadArtifactManifest, rerr error) {
+	response, err := gql.UpdateArtifactManifest(
+		as.Ctx,
+		as.GraphqlClient,
+		artifactManifestId,
+		&manifestDigest,
+		nil,
+		true,
+	)
+	if err != nil {
+		return gql.UpdateArtifactManifestUpdateArtifactManifestUpdateArtifactManifestPayloadArtifactManifest{}, err
+	}
+	return response.GetUpdateArtifactManifest().ArtifactManifest, nil
+}
+
+func (as *ArtifactSaver) updateOrCreateManifest(
+	artifactId string, baseArtifactId *string, artifactManifestId string, manifestDigest string,
+) (*string, []string, error) {
```

Comment:
This might be a case where having named return parameters both improves readability and shortens the code, since you can see what the return type is for and only set it once at the beginning for both cases (totally optional)